app/src/main/java/io/github/mattpvaughn/chronicle/features/library/LibraryViewModel.kt

```diff
@@ -168,6 +168,39 @@ class LibraryViewModel(
         }
     }
 
+    /** Compares Book titles using numerical ordering */
+    private fun sortTitleComparator(bookTitle1: String, bookTitle2: String): Int {
+        fun titleToArray(bookTitle: String): List<String> {
+            return bookTitle.split(Regex("(?<=\\D)(?=\\d)|(?<=\\d)(?=\\D)"))
+        }
+
+        val title1: Iterator<String> = titleToArray(bookTitle1).iterator()
+        val title2: Iterator<String> = titleToArray(bookTitle2).iterator()
+
+        while (true) {
```

Comment:
I get that `while(true)`s are normal in a lot of programming paradigms and perf-friendly but a little scary to me.

Some tests around this would make me less nervous. (lol sorry I know it's a little bit of a "look who's talking" given how the coverage in the app)

Reply:
@jrhedman Sorry for the delayed response again  

I've created a gist below with a sample of how I'd start approaching testing this. It mostly comes down to `TitleSortComparatorTest`, which should be mostly self explanatory. If you add it to the correct path I've added as a comment, Android Studio should be able to see that it's a test and show a green arrow by the class name to let you run the test. I also pulled `TitleSortComparator` out into it's own class to make things easier to test down the line, but that's mostly personal preference.  

You'll also need to pull down the most recent commits to `develop` on the main chronicle repo since there were some broken tests that were causing test builds to fail.

As far as approaching writing tests, that should mostly come down to gut feeling on your end, but I'd suggest you at least  cover all possible code branches in the `compare` method and maybe handle some potential real-life use cases too.

https://gist.github.com/mattttvaughn/08514c66f0bf6457564f1e8008e78db7

If you run into any issues getting the tests to run or have any questions about any of this, feel free to ping me on the development channel on the disc, happy to answer questions or help troubleshoot